extensions/src/main/java/org/sunbird/extension/user/UserExtension.java

```diff
@@ -3,16 +3,38 @@
 import java.util.Map;
 
 /**
- * UserExtension is the interface for all extensions of Sunbird user profile. It defines methods for performing CRUD operations on user profile.
- * @author Jaikumar Soundara Rajan
+ * UserExtension is the interface for all extensions of Sunbird user profile. It defines methods for
+ * performing CRUD operations on user profile.
  *
+ * @author Jaikumar Soundara Rajan
  */
 public interface UserExtension {
-	
-	/**
-	 * Creates user profile information.
-	 * @param userProfileMap User profile information
-	 */
-	public void create(Map<String,Object> userProfileMap);
-	
-}
\ No newline at end of file
+
+  /**
+   * Creates user profile information.
+   *
+   * @param userProfileMap User profile information
+   */
+  public void create(Map<String, Object> userProfileMap);
+
+  /**
+   * Reads user profile information.
+   *
+   * @param userProfileMap User profile information
+   */
+  public void read(Map<String, Object> userProfileMap);
```

Comment:
read API should accept userIdMap (User ID information) and return userProfileMap (user profile information).